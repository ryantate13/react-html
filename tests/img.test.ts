import {img} from '../src';

describe('img', () => {

    it('renders with empty attributes', () => {
        expect(img({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(img()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(img({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(img({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(img({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(img({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(img({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(img({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(img({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(img({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(img({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(img({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(img({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(img({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(img({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(img({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(img({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(img({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(img({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(img({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(img({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(img({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(img({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(img({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(img({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(img({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(img({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(img({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(img({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(img({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(img({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(img({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(img({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(img({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(img({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(img({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(img({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(img({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(img({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(img({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(img({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(img({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(img({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(img({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(img({translate: "no"})).toMatchSnapshot();
    });

    it('renders with alt="TEST"', () => {
        expect(img({alt: "TEST"})).toMatchSnapshot();
    });

    it('renders with crossOrigin="anonymous"', () => {
        expect(img({crossOrigin: "anonymous"})).toMatchSnapshot();
    });

    it('renders with crossOrigin="use-credentials"', () => {
        expect(img({crossOrigin: "use-credentials"})).toMatchSnapshot();
    });

    it('renders with decoding="sync"', () => {
        expect(img({decoding: "sync"})).toMatchSnapshot();
    });

    it('renders with decoding="async"', () => {
        expect(img({decoding: "async"})).toMatchSnapshot();
    });

    it('renders with decoding="auto"', () => {
        expect(img({decoding: "auto"})).toMatchSnapshot();
    });

    it('renders with height="1"', () => {
        expect(img({height: 1})).toMatchSnapshot();
    });

    it('renders with sizes=""', () => {
        expect(img({sizes: ""})).toMatchSnapshot();
    });

    it('renders with src="https://google.com/"', () => {
        expect(img({src: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with srcSet="es-419"', () => {
        expect(img({srcSet: "es-419"})).toMatchSnapshot();
    });

    it('renders with useMap="#foo"', () => {
        expect(img({useMap: "#foo"})).toMatchSnapshot();
    });

    it('renders with width="1"', () => {
        expect(img({width: 1})).toMatchSnapshot();
    });
});