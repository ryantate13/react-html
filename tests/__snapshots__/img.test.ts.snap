// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`img renders with accessKey="a,b,c,d" 1`] = `
<img
  accessKey="a,b,c,d"
/>
`;

exports[`img renders with alt="TEST" 1`] = `
<img
  alt="TEST"
/>
`;

exports[`img renders with autoCapitalize="characters" 1`] = `
<img
  autoCapitalize="characters"
/>
`;

exports[`img renders with autoCapitalize="none" 1`] = `
<img
  autoCapitalize="none"
/>
`;

exports[`img renders with autoCapitalize="off" 1`] = `
<img
  autoCapitalize="off"
/>
`;

exports[`img renders with autoCapitalize="on" 1`] = `
<img
  autoCapitalize="on"
/>
`;

exports[`img renders with autoCapitalize="sentences" 1`] = `
<img
  autoCapitalize="sentences"
/>
`;

exports[`img renders with autoCapitalize="words" 1`] = `
<img
  autoCapitalize="words"
/>
`;

exports[`img renders with className="foo bar baz" 1`] = `
<img
  className="foo bar baz"
/>
`;

exports[`img renders with contentEditable="false" 1`] = `
<img
  contentEditable="false"
/>
`;

exports[`img renders with contentEditable="true" 1`] = `
<img
  contentEditable="true"
/>
`;

exports[`img renders with crossOrigin="anonymous" 1`] = `
<img
  crossOrigin="anonymous"
/>
`;

exports[`img renders with crossOrigin="use-credentials" 1`] = `
<img
  crossOrigin="use-credentials"
/>
`;

exports[`img renders with decoding="async" 1`] = `
<img
  decoding="async"
/>
`;

exports[`img renders with decoding="auto" 1`] = `
<img
  decoding="auto"
/>
`;

exports[`img renders with decoding="sync" 1`] = `
<img
  decoding="sync"
/>
`;

exports[`img renders with dir="auto" 1`] = `
<img
  dir="auto"
/>
`;

exports[`img renders with dir="ltr" 1`] = `
<img
  dir="ltr"
/>
`;

exports[`img renders with dir="rtl" 1`] = `
<img
  dir="rtl"
/>
`;

exports[`img renders with draggable="false" 1`] = `
<img
  draggable="false"
/>
`;

exports[`img renders with draggable="true" 1`] = `
<img
  draggable="true"
/>
`;

exports[`img renders with empty attributes 1`] = `<img />`;

exports[`img renders with height=1 1`] = `
<img
  height={1}
/>
`;

exports[`img renders with hidden=false 1`] = `
<img
  hidden={false}
/>
`;

exports[`img renders with hidden=true 1`] = `
<img
  hidden={true}
/>
`;

exports[`img renders with id="TEST" 1`] = `
<img
  id="TEST"
/>
`;

exports[`img renders with inputMode="decimal" 1`] = `
<img
  inputMode="decimal"
/>
`;

exports[`img renders with inputMode="email" 1`] = `
<img
  inputMode="email"
/>
`;

exports[`img renders with inputMode="none" 1`] = `
<img
  inputMode="none"
/>
`;

exports[`img renders with inputMode="numeric" 1`] = `
<img
  inputMode="numeric"
/>
`;

exports[`img renders with inputMode="search" 1`] = `
<img
  inputMode="search"
/>
`;

exports[`img renders with inputMode="tel" 1`] = `
<img
  inputMode="tel"
/>
`;

exports[`img renders with inputMode="text" 1`] = `
<img
  inputMode="text"
/>
`;

exports[`img renders with inputMode="url" 1`] = `
<img
  inputMode="url"
/>
`;

exports[`img renders with is="foo" 1`] = `
<img
  is="foo"
/>
`;

exports[`img renders with itemID="https://google.com/" 1`] = `
<img
  itemID="https://google.com/"
/>
`;

exports[`img renders with itemProp="https://google.com/" 1`] = `
<img
  itemProp="https://google.com/"
/>
`;

exports[`img renders with itemRef="foo bar" 1`] = `
<img
  itemRef="foo bar"
/>
`;

exports[`img renders with itemScope=false 1`] = `
<img
  itemScope={false}
/>
`;

exports[`img renders with itemScope=true 1`] = `
<img
  itemScope={true}
/>
`;

exports[`img renders with itemType="https://google.com/" 1`] = `
<img
  itemType="https://google.com/"
/>
`;

exports[`img renders with lang="" 1`] = `
<img
  lang=""
/>
`;

exports[`img renders with lang="es-419" 1`] = `
<img
  lang="es-419"
/>
`;

exports[`img renders with sizes="" 1`] = `
<img
  sizes=""
/>
`;

exports[`img renders with slot="TEST" 1`] = `
<img
  slot="TEST"
/>
`;

exports[`img renders with spellCheck="false" 1`] = `
<img
  spellCheck="false"
/>
`;

exports[`img renders with spellCheck="true" 1`] = `
<img
  spellCheck="true"
/>
`;

exports[`img renders with src="https://google.com/" 1`] = `
<img
  src="https://google.com/"
/>
`;

exports[`img renders with srcSet="es-419" 1`] = `
<img
  srcSet="es-419"
/>
`;

exports[`img renders with style={"color":"#000000"} 1`] = `
<img
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`img renders with tabIndex=1 1`] = `
<img
  tabIndex={1}
/>
`;

exports[`img renders with title="TEST" 1`] = `
<img
  title="TEST"
/>
`;

exports[`img renders with translate="no" 1`] = `
<img
  translate="no"
/>
`;

exports[`img renders with translate="yes" 1`] = `
<img
  translate="yes"
/>
`;

exports[`img renders with useMap="#foo" 1`] = `
<img
  useMap="#foo"
/>
`;

exports[`img renders with width=1 1`] = `
<img
  width={1}
/>
`;

exports[`img renders without attributes 1`] = `<img />`;
